packages/create-turbo/src/commands/create/createProject.ts

```diff
@@ -0,0 +1,231 @@
+import retry from "async-retry";
+import chalk from "chalk";
+import fs from "fs-extra";
+import path from "path";
+import semverPrerelease from "semver/functions/prerelease";
+
+import {
+  downloadAndExtractExample,
+  downloadAndExtractRepo,
+  getRepoInfo,
+  existsInRepo,
+  hasRepo,
+  RepoInfo,
+} from "../../utils/examples";
+import { addGitIgnore } from "../../utils/git";
+import { isFolderEmpty } from "../../utils/isFolderEmpty";
+import { isWriteable } from "../../utils/isWriteable";
+import { turboLoader, error } from "../../logger";
+import cliPkgJson from "../../../package.json";
+
+export class DownloadError extends Error {}
+
+export async function createProject({
+  appPath,
+  projectName,
+  example,
+  examplePath,
+}: {
+  appPath: string;
+  projectName: string;
+  example: string;
+  examplePath?: string;
+}): Promise<{
+  cdPath: string;
+  hasPackageJson: boolean;
+  availableScripts: Array<string>;
+}> {
+  let repoInfo: RepoInfo | undefined;
+  let repoUrl: URL | undefined;
+  const isDefaultExample = example === "basic" || example === "default";
```

Comment:
Maybe an as const would be handy here

Reply:
👍🏼 moved this into a function to keep a single source of truth